src/test/kotlin/com/igorwojda/integer/factorial/challenge.kt

```diff
@@ -4,7 +4,10 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun factorial(n: Int): Int {
-    TODO("not implemented")
+    return if (n<1)
```

Comment:
Challenges - re-write this:
1. Using a folding operation: https://kotlinlang.org/docs/collection-aggregate.html#fold-and-reduce
2. Using a recursive algorithm
Please compare the running time of the two solutions.
If you're curious, look at tail recursive functions (https://kotlinlang.org/docs/reference/functions.html#tail-recursive-functions). See if you can write factorial as a tailrec function.

Reply:
After running the code after implementing the function in 3 ways (fold, classic recursion, tail recursion) and measuring the time for the execution of 10 factorial, I can say that it took:
- around 8543 nanoseconds for classic recursion
- for tail recursion the time varied a lot between 7710 nano to 13588 nano but with an average around 9763 nano
- using fold, the execution time was around 3539958 nanos